web/src/lib/components/asset-viewer/asset-viewer.svelte

```diff
@@ -447,7 +447,9 @@
 
   const handleVideoEnded = async () => {
     if ($slideshowState === SlideshowState.PlaySlideshow) {
-      await navigateAsset('next');
+      $slideshowNavigation === SlideshowNavigation.AscendingOrder
+        ? await navigateAsset('previous')
+        : await navigateAsset('next');
```

Comment:
Man, I really dislike this. We're adding more duplicate code everywhere and the asset-viewer is even more coupled to the slideshow functionality than it was before. The component should only be in charge of rendering an asset, not managing all this unrelated state.

Reply:
I moved everything to `navigateAsset`